core/exception/Exception.ts

```diff
@@ -193,7 +193,10 @@ export class Exception {
         }) as FetchException;
     }
 
-    private static createInternal(attributes: PlainObject, baseError: Error = new Error()) {
+    private static createInternal(
+        attributes: PlainObject,
+        baseError: Error = new Error(attributes.message)
```

Comment:
After reading the docs at the bottom of this page:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error
I am curious to know for what reasons our `Object.assign` approach was chosen over the 
`HoistException extends Error {` approach.

With the extends approach we would still be able to set:
this.isRoutine = false;
this.isHoistException = true;

Was the `extends` approach tried?

Reply:
I'm interested in promoting `HoistException` to its own class as well. Have you found any risks?

Curious about deleting the message key though for our current approach. I don't think that's ever necessary since `Exception.create(src: Error)` would just return (and mutate) the same error with 2 additional properties:
```
isRoutine: false,
isHoistException: true
```
No other attributes are passed in this case:
https://github.com/xh/hoist-react/blob/87173ce1b34baee1f8549e9e8e25f9a6ebbca014/core/exception/Exception.ts#L32